src/views/TriColumnListView/TriColumnListView.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import './TriColumnListView.scss';
+import ListView from '../ListView/ListView';
+import ScrollingListView from '../ScrollingListView/ScrollingListView';
+
+const prefixCls = 'kai-tricol-view';
+
+interface LocalProps {
+  onChangeIndex?: (index: number) => void,
+  focusColor?: string,
+  col1Children: any[],
+  col2Children: any[],
+  col3Children: any[],
+  onCol1ChangeIndex?: (index: number) => void,
+  onCol2ChangeIndex?: (index: number) => void,
+  onCol3ChangeIndex?: (index: number) => void,
+  selectedCol1Index?: number,
+  selectedCol2Index?: number,
+  selectedCol3Index?: number
+}
+
+const TriColListView = React.memo<LocalProps>(
+  props => {
+    const {
+      onChangeIndex,
+      focusColor,
+      col1Children,
+      col2Children,
+      col3Children,
+      onCol1ChangeIndex,
+      onCol2ChangeIndex,
+      onCol3ChangeIndex,
+      selectedCol1Index,
+      selectedCol2Index,
+      selectedCol3Index
+    } = props;
+
+    const [activeTab, setActiveTab] = useState(0);
+    const [isTransitionDone, setTransitionDone] = useState(true);
+
+    const handleChangeIndex = tabIndex => {
+      // NOTE: Ensure you set state for tab transition first.
+      //       Otherwise you will face strange race condition bugs.
+      setTransitionDone(false);
+      setActiveTab(tabIndex);
+      if(onChangeIndex) {
+        onChangeIndex(tabIndex);
+      }
+    };
+
+    
+    const handleKeyDown = useCallback(
+      e => {
+        let index = activeTab;
+        switch (e.key) {
+          case 'ArrowLeft':
+            index = index - 1 >= 0 ? index - 1 : 3 - 1;
```

Comment:
I am guessing you wrote 3 - 1 instead of 2 on purpose?

Reply:
Yeah... totally. Haha. I think it was just to illustrate the -1 but doesn't make sense in hindsight.